src/interface_ptr.rs

```diff
@@ -25,30 +28,76 @@ impl<T: ?Sized + ComInterface> InterfacePtr<T> {
     /// interface is itself trivial castable to a `*mut T::VTable`. In other words,
     /// `ptr` should also be equal to `*mut *mut T::VTable`
     ///
-    /// `ptr` must live for at least as long as the `InterfacePtr`
+    /// `ptr` must live for at least as long as the `InterfacePtr`. The underlying
+    /// COM interface is assumed to correctly implement AddRef and Release such that
+    /// the interface will be valid as long as AddRef has been called more times than
+    /// Release.
+    ///
+    /// AddRef must have been called on the underlying COM interface that `ptr` is pointing
+    /// to such that the reference count must be at least 1. It is expected that Release
+    /// will eventually be called on this pointer either manually or by passing it into
+    /// `InterfaceRc::new` which will cause Release to be called on drop of the rc.
+    ///
+    /// When this struct is dropped, `release` will be called on the underlying interface.
     ///
     /// # Panics
     ///
     /// Panics if `ptr` is null
     pub unsafe fn new(ptr: *mut *mut <T as ComInterface>::VTable) -> InterfacePtr<T> {
         InterfacePtr {
-            ptr: NonNull::new(ptr).expect("ptr was null"),
+            ptr: NonNull::new(ptr).expect("InterfacePtr's ptr was null"),
```

Comment:
I recon this was the previous behavior. Is it better to return an option rather than crashing?

Reply:
Not blocking on this though. Feel free to merge it

Reply:
I'd be fine with that. I can make that change